sljit_src/sljitNativeS390X.c

```diff
@@ -2243,28 +2243,28 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_op1(struct sljit_compiler *compile
 		}
 		/* STORE and STORE IMMEDIATE */
 		if ((dst & SLJIT_MEM) && (FAST_IS_REG(src) || src == SLJIT_IMM)) {
-			struct addr mem;
+			struct addr mem2;
```

Comment:
Hm, does this local variable even needed? It seems this was a mistake in the original code.

Reply:
Oh I see, the old code was vaguely evil. I updated all instances of `mem → mem2` within the scope of the variable... but it's sneakily accessed via `LEVAL` which is defined higher-up.

I don't really understand the original code very well, which is always a bad thing when making changes! I'll push a commit which attempts to simply remove the shadowed `struct addr mem` and see if the CI passes.

Reply:
I've thought a bit more about the code.

I think it's totally OK to remove the shadowed copies of `mem`, as I have done in the latest revision.

Reasoning: the `struct addr mem` is totally clobbered by make_addr_bxy at the start of every block of code that uses `mem`. All the fields are fully-overwritten each time it's used, so re-using a common object should not cause any problems. I can't see any gotos or other hazards.